### PR TITLE
CEDS-1749 Add indexes on declarations repository

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: test/integration/uk/gov/hmrc/exports/repositories/DeclarationRepositoryTest.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ class DeclarationRepositoryTest
       .toInt
 
   private def givenADeclarationExists(declarations: ExportsDeclaration*): Unit =
-    repository.collection.insert(false).many(declarations).futureValue
+    repository.collection.insert(ordered = true).many(declarations).futureValue
```
Comment (ed0906) [inline test/integration/uk/gov/hmrc/exports/repositories/DeclarationRepositoryTest.scala]: Why is this necessary?

File: test/util/testdata/ExportsDeclarationBuilder.scala [hunk 2]
```diff
@@ -53,7 +54,7 @@ trait ExportsDeclarationBuilder {
   )
 
   def aDeclaration(modifiers: ExportsDeclarationModifier*): ExportsDeclaration =
-    modifiers.foldLeft(modelWithDefaults)((current, modifier) => modifier(current))
+    modifiers.foldLeft(modelWithDefaults.copy(id = uuid))((current, modifier) => modifier(current))
```
Comment (ed0906) [inline test/util/testdata/ExportsDeclarationBuilder.scala]: if `modelWithDefaults` with a def then you wouldn't need this

File: test/integration/uk/gov/hmrc/exports/repositories/DeclarationRepositoryTest.scala (hunk 1 above)
Comment (bambuchaAdm) [reply test/integration/uk/gov/hmrc/exports/repositories/DeclarationRepositoryTest.scala]: yes - test depends on strict order of fetched documents - this change is only in tests

File: test/util/testdata/ExportsDeclarationBuilder.scala (hunk 2 above)
Comment (ed0906) [reply test/util/testdata/ExportsDeclarationBuilder.scala]: I'm pretty sure it used to be a `def` for this reason
